src/openrc-init/openrc-init.c

```diff
@@ -255,7 +259,10 @@ int main(int argc, char **argv)
 	}
 #endif
 
-	printf("OpenRC init version %s starting\n", VERSION);
+	quiet = rc_yesno(getenv("EINFO_QUIET"));
```

Comment:
Using `EINFO_QUIET` for this is kinda bad. Since it doesn't really have anything to do with the `einfo` stuff and also because this will be forwarded to `openrc` as well, which the user might not be wanting.

Maybe something like "OPENRC_INIT_QUIET" ?

Reply:
can one even pass environment variables to init at boot?

a kernel cmdline like `--quiet` makes sense as well imo

Reply:
the `quiet` kernel parameter is already known, and systemd even defaults to only showing errors should `quiet` be a param, so i think i makes sense to adopt that, and, if `quiet` is set we set EINFO_QUIET to true-- can't do anything about scripts that don't use einfo tho

Reply:
> the `quiet` kernel parameter is already known

The problem is, it's not passed to the init since the kernel accepted it. So you need to read `/proc/cmdline` and parse it to get it. I don't think this is worthwhile to do given PID 1 should be simple and properly parsing `/proc/cmdline` will not be super trivial. Yes, systemd does it, but I don't consider that *by itself* a convincing reason to adopt it. I'm much more open towards a `--quiet|-q` flag.

Reply:
librc already had `rc_proc_getent` that does that (used by openrc itself), and openrc-init does already link librc, so i don't see it as any increase in complexity, something like

```suggestion
    char *kernel_quiet = rc_proc_getent("quiet");
    if (kernel_quiet) {
	quiet = true;
	setenv("EINFO_QUIET", "yes", true);
    } else {
        quiet = rc_yesno(getenv("EINFO_QUIET"));
    }
    free(kernel_quiet);
```

Reply:
> librc already had `rc_proc_getent` that does that

Hmmm, didn't know that. But looking at it

https://github.com/OpenRC/openrc/blob/028eadec44f6321652fede7fb6eae77b0131bd90/src/librc/librc-misc.c#L127-L128

I see that it's doing the exact thing that I *did not* want to do: blindly splitting on spaces. If I put this into my kernel param:

```
somemod.param="one quiet two"
```

This will incorrectly pick up "quiet" from `somemod`. This is why I said *proper* `/proc/cmdline` parse will not be trivial, as you'd need to do escaping and double quote handling.

Reply:
Well, even if you fix the bug, it's still not a good idea. What if I want the kernel to be quiet but openrc to still print things?